light-client/src/operations/commit_validator.rs

```diff
@@ -53,22 +53,29 @@ impl CommitValidator for ProdCommitValidator {
         signed_header: &SignedHeader,
         validator_set: &ValidatorSet,
     ) -> Result<(), VerificationError> {
-        // TODO: `self.commit.block_id` cannot be zero in the same way as in Go
-        //       Clarify if this another encoding related issue
-        if signed_header.commit.signatures.len() == 0 {
+        let signatures = &signed_header.commit.signatures;
+
+        // Check the the commit contains at least on non-absent signature.
+        // See https://github.com/informalsystems/tendermint-rs/issues/650
+        let has_non_absent_signatures = signatures.iter().any(|cs| !cs.is_absent());
+        if !has_non_absent_signatures {
```

Comment:
My mind struggles a little to keep up with the number of logical negations going on over here 🤯 😄 

Could it perhaps be a bit more readable if it were worded like:

```suggestion
        let has_present_signatures = signatures.iter().any(|cs| !cs.is_absent());
        if !has_present_signatures {
```

or does the term "non-absent" hold special meaning above simply "present"?